qiskit/result/counts.py

```diff
@@ -54,38 +54,43 @@ def __init__(self, data, time_taken=None, creg_sizes=None,
                 ``[('c_reg', 2), ('my_creg', 4)]``.
             memory_slots (int): The number of total ``memory_slots`` in the
                 experiment.
+            include_zeros (bool): If `True`, it includes the counts in zero. Default `False`.
+
         Raises:
             TypeError: If the input key type is not an int or string
             QiskitError: If a dit string key is input with creg_sizes and/or
                 memory_slots
         """
-        bin_data = None
         data = dict(data)
-        if not data:
-            self.int_raw = {}
-            self.hex_raw = {}
-            bin_data = {}
-        else:
+
+        self.int_raw = {}
+        self.hex_raw = {}
+        bin_data = {}
+
+        if include_zeros:
+            for slot_int in range(memory_slots):
+                self.int_raw[slot_int] = 0
+                self.hex_raw[hex(slot_int)] = 0
+                slot_bin = bin(slot_int)[2:].zfill(memory_slots)
+                bin_data[postprocess._separate_bitstring(slot_bin, creg_sizes)] = 0
+
+        if data:
             first_key = next(iter(data.keys()))
             if isinstance(first_key, int):
-                self.int_raw = data
-                self.hex_raw = {
-                    hex(key): value for key, value in self.int_raw.items()}
+                self.int_raw.update(data)
+                self.hex_raw.update({hex(key): value for key, value in self.int_raw.items()})
```

Comment:
Can you not make unrelated whitespace changes here, it makes this harder to review. It took me a while to realize you were not changing the functionality of this line or any of the other ones.

Reply:
fixed in d08b384